extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -692,22 +718,12 @@ export class AmpAnalytics extends AMP.BaseElement {
     const dynamicBindings = {};
     const resourceTimingSpec = trigger['resourceTimingSpec'];
     if (resourceTimingSpec) {
-      const on = trigger['on'];
-      if (on == 'ini-load') {
-        const binding = 'RESOURCE_TIMING';
-        const analyticsVar = 'resourceTiming';
-        // TODO(warrengm): Consider limiting resource timings to avoid
-        // duplicates by excluding timings that were previously reported.
-        dynamicBindings[binding] =
-            serializeResourceTiming(resourceTimingSpec, this.win);
-        expansionOptions.vars[analyticsVar] = binding;
-      } else {
-        // TODO(warrengm): Instead of limiting resource timing to ini-load,
```

Comment:
I am worried about potential performance issue after removing the ini-load restriction. For events that can be triggered multiple times, each time analytics need to get the resourceTimeEntries and filter them by the entry's finishing time. I think we need to come up with a way to stop reporting resourceTiming after some point. (Like `responseBefore`? ) What do you think?

Reply:
This should be taken care of by unbinding the `resourceTiming` variable after the buffer limit has been reached, right?

Reply:
how is the buffer_size 150 chosen?
The buffer_size doesn't resolve my concern.
For example: If resourceTimingEntries size gets to 149 and never increase after.
Then we decide to send resourceTimingEntries with `timer` event with an interval 1 sec. 

What we will end up doing is to call `window.performance.getEntriesByType('resource')` and the `entries.filter(...)` every 1 second. Even we are not expecting new resource timing entries after say 5 seconds. And `buffer_size_limit` won't help much in this situation.

Reply:
150 is the default limit imposed by browsers according to the spec. I added a second 30 second limit in case the page doesn't hit 150 resources.

I don't think `window.performance.getEntriesByType('resource')` is particularly expensive. In local testing, it takes ~0.1ms sometimes and 0ms at other times. I'm more concerned about processing `resourceTimingSpec` in case the user provided many regexps that need to be processed. It may be worth enforcing limits on `resourceTimingSpec` in a future PR.

Reply:
Thanks. Fair enough. In that case I won't even care about the 30sec limit : )